FireboltPrivacy/FireboltPrivacyImplementation.cpp

```diff
@@ -0,0 +1,138 @@
+        FireboltPrivacyImplementation(const FireboltPrivacyImplementation&) = delete;
+        FireboltPrivacyImplementation& operator= (const FireboltPrivacyImplementation&) = delete;
+        FireboltPrivacyImplementation()
+            : _adminLock()
+            , _config()
+            , _deviceManifest()
+            , _allowResumePoints(false)
+        {
+        }
+        ~FireboltPrivacyImplementation() override
+        {
+            _deviceManifest.clear();
+            if (_service) {
+                _service->Release();
+                _service = nullptr;
+            }
+        }
+
+        uint32_t Register(Exchange::IFireboltPrivacy::INotification* notification) override
+        {
+            ASSERT(notification);
+            _adminLock.Lock();
+            notification->AddRef();
+            _notifications.push_back(notification);
+            _adminLock.Unlock();
+
+            return Core::ERROR_NONE;
+        }
+
+        uint32_t Unregister(Exchange::IFireboltPrivacy::INotification* notification) override
+        {
+            ASSERT(notification);
+            _adminLock.Lock();
+            auto item = std::find(_notifications.begin(), _notifications.end(), notification);
+            ASSERT(item != _notifications.end());
+            _notifications.erase(item);
+            (*item)->Release();
+            _adminLock.Unlock();
+
+            return Core::ERROR_NONE;
+        }
+        uint32_t AllowResumePoints(bool& allowResumePoints /* @out */) const override {
+            _adminLock.Lock();
+            allowResumePoints = _allowResumePoints;
+            _adminLock.Unlock();
+	    return 0;
+        }
+        uint32_t SetAllowResumePoints(const bool& allowResumePoints ) override {
+            _adminLock.Lock();
+            _allowResumePoints = allowResumePoints;
+	    for(auto notification: _notifications){
+	        notification->OnAllowResumePointsChanged(allowResumePoints);
+	    }
+            _adminLock.Unlock();
+	    return 0;
+        }
+        uint32_t Configure(PluginHost::IShell* service)  override {
+	    ASSERT( service != nullptr);
+            SYSLOG(Logging::Notification, (_T("service Config: %s"), service->ConfigLine().c_str()));
+            _config.FromString(service->ConfigLine());
+            _deviceManifest = _config.fireboltconfigpath.IsSet() ? _config.fireboltconfigpath.Value(): "/etc/" + _config.devicemanifest.IsSet() ? _config.devicemanifest.Value() : "firebolt-device-manifest.json" ;
+	    return 0;
+        }
+
+        BEGIN_INTERFACE_MAP(FireboltPrivacyImplementation)
+        INTERFACE_ENTRY(Exchange::IFireboltPrivacy)
+        INTERFACE_ENTRY(Exchange::IConfiguration)
+        END_INTERFACE_MAP
+
+    private:
+        mutable Core::CriticalSection _adminLock;
+        Config _config;
+        string _deviceManifest;
+        bool _allowResumePoints;
+        PluginHost::IShell* _service;
+        std::vector<Exchange::IFireboltPrivacy::INotification*> _notifications;
```

Comment:
we also use a using so it can be easily changed and used as base type for the iterators